csunplugged/topics/content/en/binary-numbers/programming-challenges/dots-largest-to-1/python-hints.md

```diff
@@ -1,7 +1,7 @@
 -   Make a variable called `number_of_dots` and set it to the largest number
     of dots entered as the input.
--   Divide the value of `number_of_dots` by 2 and display the result as the
-    output while `number_of_dots` is greater than or equal to 1.
+-   Divide the value of `number_of_dots` by 2, convert the result to an integer using `int()`
+    and display it as the output while `number_of_dots` is greater than or equal to 1.
```

Comment:
int() works here because it's definitely acting on halved powers of 2, but it's really bad practice in general. I can see users having two questions:
- *"If I halve the number 8 I get the number 4, so why do I need to convert it to an int?"*
- **After applying the solution to other situations**
*Why does 3.9 get rounded down to 3?*

So we should probably do one of:
- Don't say how to convert it, and just leave the user to figure it out themselves
- Explain that int() trims off the following decimal values, as well as why they're there
- Use // instead of / in the solution and hint, and explain what // does

Reply:
I decided to use round()